airbyte-integrations/connectors/destination-iceberg-v2/src/test-integration/kotlin/io/airbyte/integrations/destination/iceberg/v2/IcebergDestinationCleaner.kt

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.destination.iceberg.v2
+
+import io.airbyte.cdk.load.test.util.DestinationCleaner
+import io.airbyte.cdk.load.test.util.IntegrationTest.Companion.isNamespaceOld
+import io.airbyte.cdk.load.test.util.IntegrationTest.Companion.randomizedNamespaceRegex
+import org.apache.iceberg.catalog.Catalog
+import org.apache.iceberg.catalog.Namespace
+import org.apache.iceberg.catalog.SupportsNamespaces
+
+class IcebergDestinationCleaner(private val catalog: Catalog) : DestinationCleaner {
```

Comment:
do you remember why that method has an explicit `io.deletePrefix` call? afaict, `glueCatalog.dropTable(..., purge = true)` is sufficient to delete the underlying files from S3

(maybe the nessie catalog behaves differently?)

Reply:
@jdpgrailsdev would know it!

Reply:
@edgao Nessie doesn't delete the data when the drop table API is called.  This method was added to also cleanup the stored data files associated the table when we drop it via the API call.  This may or may not be necessary for all catalogs, so we should test what happens in Glue if we just call the API method on the catalog to delete the table.

Reply:
nessie why :( 

(done in https://github.com/airbytehq/airbyte/pull/49467/commits/ea13d27d23fd689280c4c4acc86b420bf644dbad)

Reply:
I confirmed that glue does the right thing if you set the `purge` flag to true, but... it's easy enough to call the table cleaner always